index.js

```diff
@@ -19,7 +17,7 @@ export default async function getStream(inputStream, options = {}) {
 	const {maxBuffer = Number.POSITIVE_INFINITY, encoding = 'utf8'} = options;
 	const isBuffer = encoding === 'buffer';
 
-	const stream = new PassThroughStream({highWaterMark: maxHighWaterMark, encoding: isBuffer ? undefined : encoding});
```

Comment:
When `highWaterMark` is set, it buffers the whole string, which makes it crash on big strings. It also prevents the `for` loop from iterating normally, as it waits for the whole input to be buffered before starting the iteration.